src/cpp/src/partition_manager.cpp

```diff
@@ -120,13 +128,49 @@ void PartitionManager::init_partitions(
     }
 }
 
+std::shared_ptr<arrow::Table> PartitionManager::filterRowById(
```

Comment:
I'm not sure the usefulness of this function. Why are we filtering on a single id value? We also already keep track of ids in the index partitions.